### PR TITLE
feat: rerank the packages in auto import

### LINKED ISSUE #9358: Refine ranking in `AutoImportCompleter`
We currently give all `AutoImportCompletion` the `Lower` ranking.

I would like to give `Low` if the completion is from one of the core packages:

- java.lang
- java.io
- java.nio
- java.util

I would like to give `Lowest` if the completion is from:

- com.sun
- sun.

Everything else stays in between at `Lower`. For example, `javax.Swing.Blah` is at `Lower`. So is `org.apache.Blah`.

### PR DESCRIPTION
**Pull request by stormckey**
fixes #9358

Also renamed `path` with namespace and qualifiedName

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/AutoImportCompleter.scala [hunk 1]
```diff
@@ -54,11 +54,35 @@ object AutoImportCompleter {
...
+    *  - java.nio
+    *  - java.util
+    *
+    * And we will give those packages a Lowest priority:
+    * - com.sun
+    * - sun.
+    *
+    * Other packages will have a Lower priority.
+    */
+  private def mkPriotiry(qualifiedName: String): Priority = {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/completion/AutoImportCompleter.scala:81]: -> priority (spelling)

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/AutoImportCompleter.scala [hunk 2]
```diff
@@ -54,11 +54,35 @@ object AutoImportCompleter {
...
+        val label = s"$className (import $qualifiedName)"
+          AutoImportCompletion(label, className, qualifiedName, ap, Some(qualifiedName), priority)
       }
     }
  }
+
+  /**
+    * Returns the priority of the completion item based on the qualified name.
+    *
+    * We will those packages a Low priority:
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/completion/AutoImportCompleter.scala:69]: We give these packages the `Low` priority:

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/AutoImportCompleter.scala [hunk 3]
```diff
@@ -54,11 +54,35 @@ object AutoImportCompleter {
...
+    *  - java.lang
+    *  - java.io
+    *  - java.nio
+    *  - java.util
+    *
+    * And we will give those packages a Lowest priority:
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/completion/AutoImportCompleter.scala:75]: We give these packages the `Lowest` priority:

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/AutoImportCompleter.scala [hunk 4]
```diff
@@ -54,11 +54,35 @@ object AutoImportCompleter {
...
+    * - com.sun
+    * - sun.
+    *
+    * Other packages will have a Lower priority.
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/completion/AutoImportCompleter.scala:79]: -> Every other package gets the `Lower` priority.

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/AutoImportCompleter.scala [hunk 5]
```diff
@@ -54,11 +54,35 @@ object AutoImportCompleter {
...
+    */
+  private def mkPriotiry(qualifiedName: String): Priority = {
+    val corePackages = List("java.lang", "java.io", "java.nio", "java.util")
+    val sunPackages = List("com.sun", "sun.")
+    if (corePackages.exists(qualifiedName.startsWith)) Priority.Low
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/completion/AutoImportCompleter.scala:84]: Add more line breaks like:

if (...)
  ...
else if (...)
  ..
else 
  ...

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/AutoImportCompleter.scala (hunk 1 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/api/lsp/provider/completion/AutoImportCompleter.scala:81]: I think in this case the string is actually `className` -- its a Java class name

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/AutoImportCompleter.scala [hunk 6]
```diff
@@ -54,11 +54,35 @@ object AutoImportCompleter {
...
+    *  - java.util
+    *
+    * And we will give those packages a Lowest priority:
+    * - com.sun
+    * - sun.
+    *
+    * Other packages will have a Lower priority.
+    */
+  private def mkPriotiry(qualifiedName: String): Priority = {
+    val corePackages = List("java.lang", "java.io", "java.nio", "java.util")
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/completion/AutoImportCompleter.scala:82]: corepackages -> frequentlyUsedPackages

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/AutoImportCompleter.scala [hunk 7]
```diff
@@ -54,11 +54,35 @@ object AutoImportCompleter {
...
+    val sunPackages = List("com.sun", "sun.")
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/completion/AutoImportCompleter.scala:83]: -> rarelyUsedPackages

**Review by magnus-madsen — COMMENTED** [review]
lots of minor things

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/AutoImportCompleter.scala (hunk 1 above)
Comment (stormckey) [reply main/src/ca/uwaterloo/flix/api/lsp/provider/completion/AutoImportCompleter.scala:81]: > I think in this case the string is actually `className` -- its a Java class name

Okay we can change to that, but in the code above we use className to refer to `Math` and qualifiedName to refer to `java.lang.Math`. Maybe we should be consistent? Or other names?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/api/lsp/provider/completion/AutoImportCompleter.scala]: I see.

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**